### PR TITLE
Fixes #490 Open editImage on taking picture

### LINKED ISSUE #490: Open Edit image activity on clicking photo
**Actual Behaviour**

After taking the picture from the camera, it just saves the image.

**Expected Behaviour**

User should be taken to Editing image activity after taking the picture.

**Steps to reproduce it**

Open camera
Take Picture

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by anantprsd5**
Fixes issue #490 Open editImage Activity on taking picture

Changes: Added callback to open Photo activity with three options to save, edit and retake photo.
Here is the screen shot for the same.
photo_2017-06-02_21-33-17

### CODE REVIEW

**Comment by vinaysajjanapu** [comment]
@anantprsd5 Wouldn't it be better if you add a page in between to display the captured image and ask user whether to retake the picture or proceed to editing

**Comment by anantprsd5** [comment]
@vinaysajjanapu Yes. Will make the changes soon.

**Comment by anantprsd5** [comment]
@vinaysajjanapu @mohitmanuja @pa1pal @subhankar29 I just checked with the other apps like whatsapp and instagram, they are using the back button to delete the clicked image and to go to camera activity to take the picture again, we can also do it similarly. I don't feel the need of another page for it. Please give suggestions on this.

**Comment by mohitmanuja** [comment]
@anantprsd5 Adding page will be good. We can do something like this 

screenshot_2017-06-01-18-12-36-789_com magicv airbrush

Add a page that contains three buttons at the bottom.
1. Back ( Open camera again without saving image)
2. Save ( Save the image and open camera again)
3. Edit ( Open EditImageActivity)
 
Suggestions required.

**Comment by Subhankar29** [comment]
I agree with @mohitmanuja, it can save the users from confusions.

**Comment by anantprsd5** [comment]
Sure then. Adding it now.

**Comment by anantprsd5** [comment]
@vinaysajjanapu Done. Thanks.

**Comment by anantprsd5** [comment]
@pa1pal Can you please review it if the camera is fixed?

File: app/src/main/res/layout/activity_photo.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom"
+        android:background="@color/white">
+
+        <ImageButton
+            android:id="@+id/delete"
+            android:layout_width="80dp"
+            android:layout_height="80dp"
+            android:layout_alignParentLeft="true"
+            android:src="@drawable/ic_close_black_24dp"
```
Comment (pa1pal) [inline app/src/main/res/layout/activity_photo.xml]: use srcCompat.

File: app/src/main/res/layout/activity_photo.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    android:layout_width="fill_parent"
+    android:layout_height="fill_parent"
+    android:orientation="vertical"
+    android:background="@color/white">
+
+    <ImageView
+        android:id="@+id/imageView"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:scaleType="fitStart"
```
Comment (pa1pal) [inline app/src/main/res/layout/activity_photo.xml]: photo_view_width

Is `scaleType` causing this?

File: app/src/main/java/vn/mbm/phimp/me/opencamera/Camera/PhotoActivity.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    public void editImage(View v){
+        EditImageActivity.start(this, FILE_PATH, FileUtils.genEditFile().getAbsolutePath(),1);
+    }
+
+    public void saveOriginal(View v){
+        delete = false;
+        Intent intent = new Intent(REVIEW_ACTION, Uri.fromFile(new File(FILE_PATH)));
+        intent.setClass(getApplicationContext(), SingleMediaActivity.class);
+        this.startActivity(intent);
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/opencamera/Camera/PhotoActivity.java]: finish activity.
It is not destroying, you can check by back press after saving an image.

**Comment by anantprsd5** [comment]
@pa1pal @mohitmanuja Please check now. Made the required changes.

**Comment by anantprsd5** [comment]
screenshot_20170603-222725

**Pull request merged**